src/NuGet.Core/NuGet.Build.Tasks.Console/MSBuildStaticGraphRestore.cs

```diff
@@ -104,6 +104,13 @@ public async Task<bool> RestoreAsync(string entryProjectFilePath, IDictionary<st
                 return false;
             }
 
+            if (string.Equals(Path.GetExtension(entryProjectFilePath), ".sln", StringComparison.OrdinalIgnoreCase)
+                    && dependencyGraphSpec.Restore.Count == 0)
+            {
+                MSBuildLogger.LogInformation(string.Format(CultureInfo.CurrentCulture, Strings.Log_NoProjectsForRestore));
```

Comment:
Is this something that should maybe be logged in the BuildTasksUtility instead?

Is the behavior among static and non-static graph equivalent?

Reply:
`BuildTasksUtility` currently doesn't log any message when there are no projects to restore.
https://github.com/NuGet/NuGet.Client/blob/a5d5e3236e86f6c3c09097b723ce2b52852a1752/src/NuGet.Core/NuGet.Build.Tasks/BuildTasksUtility.cs#L234

AFAIK, `dependencyGraphSpec` model doesn't know whether `solution file` or `project file` passed to restore command. `BuildTasksUtility` also doesn't have access to the `entryProjectFilePath` argument passed to static graph restore.

Hence this log message can't be added in `BuildTasksUtility` for non-static graph equivalent restore (for e.g. dotnet restore). I could be wrong but happy to learn more incase if I am missing something.